Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,27 +138,64 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    auto result = apply_settings_impl();
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), result.release_error().string_literal());
+        return;
+    }
+}
+
+ErrorOr<void> NetworkSettingsWidget::apply_settings_impl()
```

Comment:
nit: i feel this one deserves more the name of `apply_settings` (it applies and possibly returns an error); iʼve got no proper naming for the previous method, though.